src/Illuminate/Database/Eloquent/FactoryBuilder.php

```diff
@@ -125,21 +125,29 @@ public function make(array $attributes = [])
      */
     protected function makeInstance(array $attributes = [])
     {
-        return Model::unguarded(function () use ($attributes) {
-            if (! isset($this->definitions[$this->class][$this->name])) {
-                throw new InvalidArgumentException("Unable to locate factory with name [{$this->name}] [{$this->class}].");
-            }
+        $attributes = array_merge($this->buildDefinition($attributes), $attributes);
 
-            $definition = call_user_func(
+        return new $this->class($this->callClosureAttributes($attributes));
```

Comment:
The model's class name is stored in `$this->class`.

So this line actually creates a new model instance and returns it.